Unable to deploy via Actions: Cannot read property 'name' of null

Trying to use this action in my personal project (private repo so far). YML file is as follows: https://gist.github.com/pvasilyev/ceaf81d6943464bf77ac2a6a0b7602ad

I've verified that my aws user has all permissions listed in the https://github.com/aws-actions/amazon-ecs-deploy-task-definition#permissions but still w/o luck.

Logs excerpt:

```
Run aws-actions/amazon-ecs-deploy-task-definition@v1
  with:
    task-definition: /home/runner/work/_temp/task-definition-3216nPr47VbD5lae.json
    service: tgbot-webhook-service
    cluster: tgbot-springboot-cluster
    wait-for-service-stability: true
  env:
    AWS_DEFAULT_REGION: us-east-1
    AWS_REGION: us-east-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
##[error]Cannot read property 'name' of null
```

--- a/index.js
+++ b/index.js
@@ -70,11 +70,25 @@ function isEmptyValue(value) {
     return true;
   }
 
-  if (Array.isArray(value) && value.length === 0) {
+  if (Array.isArray(value)) {
+    for (var element of value) {
+      if (!isEmptyValue(element)) {
+        // the array has at least one non-empty element
+        return false;
+      }
+    }
+    // the array has no non-empty elements
     return true;
   }
 
-  if (typeof value === 'object' && Object.values(value).length === 0) {
+  if (typeof value === 'object') {
+    for (var childValue of Object.values(value)) {
+      if (!isEmptyValue(childValue)) {
+        // the object has at least one non-empty property
+        return false;
+      }
+    }
+    // the object has no non-empty property
     return true;
   }
 
@@ -86,15 +100,8 @@ function emptyValueReplacer(_, value) {
     return undefined;
   }
 
-  if (typeof value === 'object') {
-    for (var childValue of Object.values(value)) {
-      if (!isEmptyValue(childValue)) {
-        // the object has at least one non-empty property
-        return value;
-      }
-    }
-    // the object has no non-empty property
-    return undefined;
+  if (Array.isArray(value)) {
+    return value.filter(e => !isEmptyValue(e));
   }
 
   return value;

--- a/index.test.js
+++ b/index.test.js
@@ -198,7 +198,37 @@ describe('Deploy to ECS', () => {
                 throw new Error(`Wrong encoding ${encoding}`);
             }
 
-            return '{ "memory": "", "containerDefinitions": [ { "name": "sample-container", "logConfiguration": {}, "repositoryCredentials": { "credentialsParameter": "" }, "cpu": 0, "essential": false } ], "requiresCompatibilities": [ "EC2" ], "family": "task-def-family" }';
+            return `
+            {
+                "memory": "",
+                "containerDefinitions": [ {
+                    "name": "sample-container",
+                    "logConfiguration": {},
+                    "repositoryCredentials": { "credentialsParameter": "" },
+                    "command": [
+                        ""
+                    ],
+                    "environment": [
+                        {
+                            "name": "hello",
+                            "value": "world"
+                        },
+                        {
+                            "name": "",
+                            "value": ""
+                        }
+                    ],
+                    "secretOptions": [ {
+                        "name": "",
+                        "valueFrom": ""
+                    } ],
+                    "cpu": 0,
+                    "essential": false
+                } ],
+                "requiresCompatibilities": [ "EC2" ],
+                "family": "task-def-family"
+            }
+            `;
         });
 
         await run();
@@ -209,7 +239,11 @@ describe('Deploy to ECS', () => {
                 {
                     name: 'sample-container',
                     cpu: 0,
-                    essential: false
+                    essential: false,
+                    environment: [{
+                        name: 'hello',
+                        value: 'world'
+                    }]
                 }
             ],
             requiresCompatibilities: [ 'EC2' ]

